pkg/types/vsphere/validation/platform.go

```diff
@@ -38,6 +40,24 @@ func ValidatePlatform(p *vsphere.Platform, fldPath *field.Path) field.ErrorList
 	// If all VIPs are empty, skip IP validation.  All VIPs are required to be defined together.
 	if strings.Join([]string{p.APIVIP, p.IngressVIP}, "") != "" {
 		allErrs = append(allErrs, validateVIPs(p, fldPath)...)
+		if network != nil {
+			apiVIPOverlap := false
+			ingressVIPOverlap := false
+			for _, machine := range network.MachineNetwork {
+				if machine.CIDR.Contains(net.ParseIP(p.APIVIP)) {
```

Comment:
Do all of this contain checking in `validateVIPs`. That function is already parsing the string into an IP, albeit within the call to `validate.IP`.

This makes it a bit more complicated for the call to `validateVIPs` from `ValidateForProvisioning`. But I would argue that `ValidateForProvisioning` should not be doing full validation of the VIPs. It should only be validating that the VIPs have been set. If the VIPs are set, then the full validation would have been done doing the initial validation of the install config. If the VIPs are not set, then `validateVIPs` will fail and will not have anything more to validate beyond that the VIPs are missing anyway. What do you think, @patrickdillon (https://github.com/openshift/installer/pull/3372)?

Reply:
> This makes it a bit more complicated for the call to `validateVIPs` from `ValidateForProvisioning`. But I would argue that `ValidateForProvisioning` should not be doing full validation of the VIPs. It should only be validating that the VIPs have been set. If the VIPs are set, then the full validation would have been done doing the initial validation of the install config. If the VIPs are not set, then `validateVIPs` will fail and will not have anything more to validate beyond that the VIPs are missing anyway. What do you think, @patrickdillon (#3372)?

I agree we should move this logic to `validateVIPs`. And if I understand correctly, I am fine to refactor `ValidateForProvisioning` so it only checks for the presence of the required VIPs fields and does not do the full validation.

To frame the discussion a bit more for @rna-afk vSphere UPI was released before IPI. When UPI was released VIPs were not required in the platform, but they are required for IPI. We cannot require VIPs in the platform as it would break backwards compatibility with UPI. Therefore we added the check here in `ValidateForProvisioning` to distinguish the validation only run for IPI. 

If I understand @staebler's concern correctly, we want to avoid a slippery slope of allowing validation creeping in here, where it does not really belong.